### PR TITLE
Tavis CI with build of boost on linux

### PR DESCRIPTION
**Pull request by artzet-s**
Not on OSX because it's fail.

You can look build error on : https://travis-ci.org/artzet-s/openalea-recipes/jobs/300247181

### CODE REVIEW

**Comment by artzet-s** [comment]
@pradal @pfernique

**Comment by pradal** [comment]
Thanks Simon,
Do you think that these modifications will allow to build PlantGL?
I will Merge this PR when this is True.
I do not see the package on anaconda/openalea in the label unstable? Is it normal?

**Comment by artzet-s** [comment]
> Do you think that these modifications will allow to build PlantGL?
I'm currently trying to determine that on my github fork and my anaconda channel. 
 
> I do not see the package on anaconda/openalea in the label unstable? Is it normal?

Yes, it'll be uploaded only when the PR will be merged.

**Comment by pradal** [comment]
So you need that the PR be uploaded? 
However, it will broke other recipes. Is it what is needed?

**Comment by artzet-s** [comment]
> So you need that the PR be uploaded?

No, not now. I'm agree with you to merge only when this PR allow to build plantgl.
I temporary use my anaconda channel "artzet_s" (https://anaconda.org/artzet_s)

> However, it will broke other recipes. Is it what is needed?

No, not now.

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+  - docker
+
+env:
+ - CONDA_RECIPE=boost
+   CONDA_VERSION=2
+   ANACONDA_UPLOAD=openalea
+   ANACONDA_LABEL=unstable
+
+install:
+  - git clone https://github.com/OpenAlea/travis-ci.git
```
Comment (pradal) [inline .travis.yml]: Change the url to
https://github.com/openalea/travis-ci.git

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+deploy:
+  skip_cleanup: true
+  provider: script
+  script: bash deploy_script.sh
+
+after_deploy:
+  - source after_deploy.sh
+
+after_script:
+  - source after_script.sh
```
Comment (pradal) [inline .travis.yml]: Add a new line

**Comment by pradal** [comment]
I have seen that you upload the recipe on openalea/label/unstable.
So it is OK for me to merge the PR.

**Comment by pradal** [comment]
Update the version of Boost in line with the one used in conda-forge (1.57 -> 1.64?)

**Comment by artzet-s** [comment]
Ok thanks for your's comment review . 

I updated my code. Unfortunately for the moment, it doesn't work with plantgl installation. 
I have a -lboost_system error. `/usr/bin/ld: cannot find -lboost_system`

https://travis-ci.org/artzet-s/plantgl/builds/301316486

> Update the version of Boost in line with the one used in conda-forge (1.57 -> 1.64?)

Don't know. Currently, my main focus is to have something working on conda, unless a more recent version fix building bug, i prefer, do that in another PR.

**Comment by pradal** [comment]
Thanks for your quick answer. Where is your PlantGL recipe? On which repo?
OK for delaying the boost version modif in another PR

**Comment by pradal** [comment]
You can look at my branch:
https://github.com/pradal/plantgl/tree/travis_conda

I have added a conda subdir that is updated. Moreover, you have to update the travis.yaml file to the new system

**Comment by pradal** [comment]
There is a PR:
https://github.com/VirtualPlants/plantgl/pull/4/files

**Comment by artzet-s** [comment]
Thanks to @pfernique for debugging me on the boost build. CXX and LINKS flags was missing on linux build.

I increase the travis wait time (60) cause, sometime is still fail on osx build (on my travis channel).

> Yes. sysctl -n hw.ncpu is the only reliable way to use CPU_COUNT on Mac.

For CPU_COUNT or sysctl -n hw.ncpu, I don't saw much difference. Maybe like travis use docker and CPU_COUNT variable is well setted.
 
Nevertheless, for time processing, It's weird to me, cause on linux, boost build in ~20 minutes each time, and on osx 35/+45 minutes. 

Any hint to explain why ? (Or just gcc > clang * 2 ? linux > osx * 2 ? )

@pfernique have you the same issue ?

**Comment by pradal** [comment]
Thks.

**Comment by pfernique** [comment]
Look at https://docs.travis-ci.com/user/reference/overview/
As you can see build environments are not the same for linux and osx.

**Comment by artzet-s** [comment]
I made a mistake, can you re-open PR or change that : 

`ANACONDA_UPLOAD=artzet_s
 ANACONDA_LABEL=main`

by : 

`ANACONDA_UPLOAD=openalea
 ANACONDA_LABEL=unstable`

**Comment by pradal** [comment]
Done

**Pull request merged**